cpp/ScanChain_Hgg.C

```diff
@@ -409,6 +438,66 @@ int ScanChain_Hgg(TChain *ch, double genEventSumw, TString year, TString process
 
       if (dijets[0].p4.M()<50) continue;
 
+      if (isMC){
+        LeadPhoton_genPartFlav = int(selectedDiPhoton.leadPho.genPartFlav());
+        SubleadPhoton_genPartFlav = int(selectedDiPhoton.subleadPho.genPartFlav());
+        TLorentzVector GenHiggs, GenX, GenY;
+        GenParts genparts = getGenParts();
+        vector<TLorentzVector> gen_child_xyh, gen_child_ygg, gen_child_hbb;
+        for (int igenpart=0; igenpart<genparts.size(); igenpart++)
+          {
+            if (genparts[igenpart].isxyh()) {
+              GenX = genparts[igenpart].mother_p4(); 
+              gen_child_xyh.push_back(genparts[igenpart].p4());
+              GenX_pt = GenX.Pt();
+              GenX_eta = GenX.Eta();
+              GenX_phi = GenX.Phi();
+              GenX_mass = GenX.M();
+            }
+            if (genparts[igenpart].isygg()) {
+              GenY = genparts[igenpart].mother_p4(); 
+              gen_child_ygg.push_back(genparts[igenpart].p4());
+              GenY_pt = GenY.Pt();
+              GenY_eta = GenY.Eta();
+              GenY_phi = GenY.Phi();
+              GenY_mass = GenY.M();
+            }
+            if (genparts[igenpart].ishbb()) {
+              GenHiggs = genparts[igenpart].mother_p4();
+              gen_child_hbb.push_back(genparts[igenpart].p4());
+              GenHiggs_pt = GenHiggs.Pt();
+              GenHiggs_eta = GenHiggs.Eta();
+              GenHiggs_phi = GenHiggs.Phi();
+              GenHiggs_mass = GenHiggs.M();
+            }
+          }
+        TLorentzVector sort_GenPart;
+//          if (gen_child_xyh[0].Pt()<gen_child_xyh[1].Pt()) {sort_GenPart = gen_child_xyh[0]; gen_child_xyh[0]= gen_child_xyh[1]; gen_child_xyh[1] = sort_GenPart;}
+//          if (gen_child_ygg[0].Pt()<gen_child_ygg[1].Pt()) {sort_GenPart = gen_child_ygg[0]; gen_child_ygg[0]= gen_child_ygg[1]; gen_child_ygg[1] = sort_GenPart;}
```

Comment:
Are these lines commented out on purpose or by mistake? If on purpose, what's the point of defining `sort_GenPart` in the previous line?

Then I saw that you are using this for the Hbb case, so maybe indeed the lines were comment out on purpose, so why not delete them, if not needed? In any case, the variable need to be defined in a more relevant scope (in the `if` statement it is used) and with a more descriptive name (I guess it is a temporary variable for rearranging TLorentzVectors, so `gen_child_hbb_temp`?).